src/xenia/vfs/devices/stfs_container_device.cc

```diff
@@ -207,76 +220,99 @@ StfsContainerDevice::Error StfsContainerDevice::ReadSVOD() {
-      svod_layout_ = SvodLayoutType::kSingleFile;
-      XELOGI("SVOD is a single file. Magic block present at 0xD000.");
+  } else {
+    xe::filesystem::Seek(svod_header, 0x12000, SEEK_SET);
+    fread(magic_buf, 1, 20, svod_header);
+    if (memcmp(magic_buf, MEDIA_MAGIC, 20) == 0) {
+      // If the SVOD's magic block is at 0x12000, it is likely using an XSF
+      // layout. This is usually due to converting the game using a third-party
+      // tool, as most of them use a nulled XSF as a template.
+
+      svod_base_offset_ = 0x10000;
+      magic_offset = 0x12000;
+
+      // Check for XSF Header
+      const char* XSF_MAGIC = "XSF";
+      xe::filesystem::Seek(svod_header, 0x2000, SEEK_SET);
+      fread(magic_buf, 1, 3, svod_header);
+      if (memcmp(magic_buf, XSF_MAGIC, 3) == 0) {
+        svod_layout_ = SvodLayoutType::kXSF;
+        XELOGI("SVOD uses an XSF layout. Magic block present at 0x12000.");
+        XELOGI("Game was likely converted using a third-party tool.");
+      } else {
+        svod_layout_ = SvodLayoutType::kUnknown;
+        XELOGI("SVOD appears to use an XSF layout, but no header is present.");
+        XELOGI("SVOD magic block found at 0x12000");
+      }
     } else {
-      svod_layout_ = SvodLayoutType::kUnknown;
-      XELOGE(
-          "SVOD is not a single file, but the magic block was found at "
-          "0xD000.");
+      xe::filesystem::Seek(svod_header, 0xD000, SEEK_SET);
+      fread(magic_buf, 1, 20, svod_header);
+      if (memcmp(magic_buf, MEDIA_MAGIC, 20) == 0) {
+        // If the SVOD's magic block is at 0xD000, it most likely means that it
+        // is a single-file system. The STFS Header is 0xB000 bytes , and the
+        // remaining 0x2000 is from hash tables. In most cases, these will be
+        // STFS, not SVOD.
+
+        svod_base_offset_ = 0xB000;
+        magic_offset = 0xD000;
+
+        // Check for single file system
+        if (header_.metadata.data_file_count == 1) {
+          svod_layout_ = SvodLayoutType::kSingleFile;
+          XELOGI("SVOD is a single file. Magic block present at 0xD000.");
+        } else {
+          svod_layout_ = SvodLayoutType::kUnknown;
+          XELOGE(
+              "SVOD is not a single file, but the magic block was found at "
+              "0xD000.");
+        }
+      } else {
+        XELOGE("Could not locate SVOD magic block.");
+        return Error::kErrorReadError;
+      }
     }
-  } else {
-    XELOGE("Could not locate SVOD magic block.");
-    return Error::kErrorReadError;
   }
 
   // Parse the root directory
-  uint8_t* magic_block = data + magic_offset_;
-  uint32_t root_block = xe::load<uint32_t>(magic_block + 0x14);
-  uint32_t root_size = xe::load<uint32_t>(magic_block + 0x18);
-  uint32_t root_creation_date = xe::load<uint32_t>(magic_block + 0x1C);
-  uint32_t root_creation_time = xe::load<uint32_t>(magic_block + 0x20);
+  xe::filesystem::Seek(svod_header, magic_offset + 0x14, SEEK_SET);
+
+  uint32_t root_block;
```

Comment:
Perhaps a locally defined struct to read into, as opposed to a batch set of variables?